Source/WebCore/loader/MixedContentChecker.cpp

```diff
@@ -184,11 +190,11 @@ bool MixedContentChecker::shouldUpgradeInsecureContent(LocalFrame& frame, IsUpgr
         // 4.1.2 request’s URL’s host is an IP address.
         || (!shouldUpgradeIPAddressAndLocalhostForTesting && URL::hostIsIPAddress(url.host()))
         // 4.1.4 request’s destination is not "image", "audio", or "video".
-        || (destination != FetchOptions::Destination::Audio && destination != FetchOptions::Destination::Image && destination != FetchOptions::Destination::Video)
+        || (!destinationIsImageAudioOrVideo(destination))
         // 4.1.5 request’s destination is "image" and request’s initiator is "imageset".
         || (destination == FetchOptions::Destination::Image && initiator == Initiator::Imageset)
         // and CORS is excluded
-        || mode == FetchOptions::Mode::Cors)
+        || (mode == FetchOptions::Mode::Cors && (!document->quirks().needsRelaxedCorsMixedContentCheckQuirk() || !destinationIsImageAudioOrVideo(destination))))
```

Comment:
Or we could just remove this because we're essentially ignoring the fetch mode and we shouldn't reach this condition if we already checked `!destinationIsImageAudioOrVideo` and that was true. Maybe just add a FIXME and we should bring the up with WebAppSec.

Reply:
I think we do want this (unless the Fetch mode is impractical given the state of the web today). My reasoning is that some sites have Fetch::mode set properly and want it enforced, and we should honor it.

This change just says:
"If a site has Cors set, if its one of the (currently one) site(s) that break if we do this quirk for image/audio/media, allow the upgrade. Otherwise, do what the site author wanted."

Another possibility would be for me to pass the Fetch Mode to the Quirk for checking, but that would involve telling Quirks.cpp about FetchOptions, which just adds more include complexity. So I thought this check might be simpler.

Reply:
@sysrqb I revised line 197 slightly to make it clearer that we are checking:
(1) Fetch mode is one where we are supposed to reject upgrading
(2) -AND- we are NOT in the special state of:
(a) We are on a site needing the Quirk
(b) The load is also for image/audio/media